src/test/java/org/repodriller/persistence/csv/CSVFileTest.java

```diff
@@ -0,0 +1,32 @@
+package org.repodriller.persistence.csv;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.repodriller.RepositoryMining;
+import org.repodriller.filter.range.Commits;
+import org.repodriller.scm.GitRepository;
+
+public class CSVFileTest {
+
+
+	private String pathToRepo7;
+
+	@Before
+	public void setUp() {
+		this.pathToRepo7 = this.getClass().getResource("/").getPath() + "../../test-repos/git-7";
+	}
+	
+	@Test(expected = CSVFileFormatException.class)
+		public void createCSV() {
+		TestCSVisitor visitor = new TestCSVisitor();
+		CSVFile file = new CSVFile("C:/Users/dev/Desktop","test.csv", new String [] {"Hash"});
+		
+		new RepositoryMining()
+			.in(GitRepository.singleProject(pathToRepo7))
+			.through(Commits.all())
+			.process(visitor, file)
+			.mine();
+		}
+		
```

Comment:
you should also test:
- One single column working
- Many columns working
- One single column failing
- Many columns failing

To test it, you will have to really write the file somewhere and open it to see if it has the structure you want. I suggest you to use `String path = System.getProperty("java.io.tmpdir") + "file.csv")` to get the path to a temporary directory. Then, every time the test runs (`@Before`), you delete this file. 

Does that make sense to you?